pkg/kinds/general.go

```diff
@@ -0,0 +1,131 @@
+package kinds
+
+import (
+	"time"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ResourceOriginInfo is saved in annotations
+type ResourceOriginInfo struct {
+	Name      string     `json:"name,omitempty"`
+	Path      string     `json:"path,omitempty"`
+	Key       string     `json:"key,omitempty"`
+	Timestamp *time.Time `json:"time,omitempty"`
```

Comment:
some inline docs describing each of these fields would be useful I think

Reply:
In various projects and the POC this mapped to 
<img width="827" alt="image" src="https://github.com/grafana/grafana/assets/705951/5ef7cf26-4a7f-4036-848c-463d588432dd">